Test/Tests/Network/DNS.Tests.ps1

```diff
@@ -0,0 +1,395 @@
+. $PSScriptRoot\..\..\PesterLogger\RemoteLogCollector.ps1
+
+. $PSScriptRoot\..\..\TestConfigurationUtils.ps1
+. $PSScriptRoot\..\..\Utils\NetAdapterInfo\RemoteContainer.ps1
+. $PSScriptRoot\..\..\Utils\ContrailNetworkManager.ps1
+. $PSScriptRoot\..\..\Utils\MultiNode\ContrailMultiNodeProvisioning.ps1
+
+. $PSScriptRoot\..\..\Utils\ContrailAPI\DNSServerRepo.ps1
+. $PSScriptRoot\..\..\Utils\ContrailAPI\IPAMRepo.ps1
+
+$ContainersIDs = @("jolly-lumberjack","juniper-tree")
+
+$Subnet = [SubnetConfiguration]::new(
+    "10.0.5.0",
+    24,
+    "10.0.5.1",
+    "10.0.5.19",
+    "10.0.5.83"
+)
+$Network = [Network]::New("testnet12", $Subnet)
+
+$TenantDNSServerAddress = "10.0.5.80"
+
+$VirtualDNSServer = [DNSServer]::New("CreatedForTest")
+
+$VirtualDNSrecords = @([DNSRecord]::New("vdnsrecord-nonetest", "1.1.1.1", "A", $VirtualDNSServer.GetFQName()),
+                       [DNSRecord]::New("vdnsrecord-defaulttest", "1.1.1.2", "A", $VirtualDNSServer.GetFQName()),
+                       [DNSRecord]::New("vdnsrecord-virtualtest", "1.1.1.3", "A", $VirtualDNSServer.GetFQName()),
+                       [DNSRecord]::New("vdnsrecord-tenanttest", "1.1.1.4", "A", $VirtualDNSServer.GetFQName())
+)
+
+$DefaultDNSrecords = @([DNSRecord]::New("defaultrecord-nonetest.com", "3.3.3.1", "A", ""),
+                       [DNSRecord]::New("defaultrecord-defaulttest.com", "3.3.3.2", "A", ""),
+                       [DNSRecord]::New("defaultrecord-virtualtest.com", "3.3.3.3", "A", ""),
+                       [DNSRecord]::New("defaultrecord-tenanttest.com", "3.3.3.4", "A", "")
+)
+
+function Create-Container {
+    Param (
+        [Parameter(Mandatory=$true)] [PSSessionT] $Session,
+        [Parameter(Mandatory=$true)] [string] $ContainerID,
+        [Parameter(Mandatory=$true)] [string] $ContainerImage,
+        [Parameter(Mandatory=$true)] [string] $NetworkName,
+        [Parameter(Mandatory=$false)] [string] $IP
+    )
+
+    Write-Log "Creating container: $ContainerID"
+    New-Container `
+        -Session $Session `
+        -NetworkName $NetworkName `
+        -Name $ContainerID `
+        -Image $ContainerImage `
+        -IP $IP
+
+    Write-Log "Getting container' NetAdapter Information"
+
+    $ContainerNetInfo = Get-RemoteContainerNetAdapterInformation `
+        -Session $Session -ContainerID $ContainerID
+    $IP = $ContainerNetInfo.IPAddress
+    Write-Log "IP of $ContainerID : $IP"
+
+    return $IP
+}
+
+# This function is used only to generate command
+# that will be passed to docker exec.
+# $Hostname will be substituted.
+function Resolve-DNSLocally {
+    $resolved = (Resolve-DnsName -Name $Hostname -Type A -ErrorAction SilentlyContinue)
+
+    if($error.Count -eq 0) {
+        Write-Host "found"
+        $resolved[0].IPAddress
+    } else {
+        Write-Host "error"
+        $error[0].CategoryInfo.Category
+    }
+}
+$ResolveDNSLocallyCommand = (${function:Resolve-DNSLocally} -replace "`n|`r", ";")
```

Comment:
Can't we move into body of Resolve-DNS?

Reply:
Because it is as constant in whole script, so I didn't want to declare a function and create a string out of it every time I resolve. I moved this part to "declaration area" at the beginning of a script.